drivers/dac/dac_sam.c

```diff
@@ -25,9 +25,9 @@
 #include <zephyr/irq.h>
 LOG_MODULE_REGISTER(dac_sam, CONFIG_DAC_LOG_LEVEL);
 
-BUILD_ASSERT(IS_ENABLED(CONFIG_SOC_SERIES_SAME70) ||
-	     IS_ENABLED(CONFIG_SOC_SERIES_SAMV71),
-	     "Only SAME70, SAMV71 series devices are currently supported.");
+BUILD_ASSERT(IS_ENABLED(CONFIG_SOC_SERIES_SAME70) || IS_ENABLED(CONFIG_SOC_SERIES_SAMV71) ||
+		     IS_ENABLED(CONFIG_SOC_SERIES_SAM4S),
+	     "Only SAME70, SAMV71, SAM4S series devices are currently supported.");
```

Comment:
With your patch all SAM is supported, unless SAM4L, which requires another devicetree bind. So this is not necessary anymore : )